docs: enhance documentation (#191)
elastic/e2e-testing: e2e/_suites/helm/README.md

> [![Build Status](https://beats-ci.elastic.co/buildStatus/icon?job=e2e-tests%2Fe2e-testing-helm-daily)](https://beats-ci.elastic.co/job/e2e-tests/job/e2e-testing-helm-daily/)

Reviewer: As long as the daily job does not wait for the downstream job's status then let's use the downstream one:

```suggestion
[![Build Status](https://beats-ci.elastic.co/buildStatus/icon?job=e2e-tests%2Fe2e-testing-mbp%2Fmaster)](https://beats-ci.elastic.co/job/e2e-tests/job/e2e-testing-mbp/job/master/)
```

# Observability Helm charts End-To-End tests

## Motivation

Our goal is for the Observability team to execute this automated e2e test suite while developing the Helm charts for APM Server, Filebeat and Metricbeat. The tests in this folder assert that the use cases (or scenarios) defined in the `features` directory are behaving as expected.

## How do the tests work?

At the topmost level, the test framework uses a BDD framework written in Go, where we set
the expected behavior of use cases in a feature file using Gherkin, and implementing the steps in Go code.

> The provisining of services is accomplish using [Kind (Kubernetes in Docker)](https://kind.sigs.k8s.io/https://kind.sigs.k8s.io/) and [Helm](https://helm.sh/) packages.

Reviewer: ```suggestion
The provisioning of services is accomplished using [Kind (Kubernetes in Docker)](https://kind.sigs.k8s.io/https://kind.sigs.k8s.io/) and [Helm](https://helm.sh/) packages.
```

### Running the tests

1. Clone this repository, say into a folder named `e2e-testing`.

   ``` shell
   git clone git@github.com:elastic/e2e-testing.git
   cd e2e-testing/e2e/_suites/helm
   ```

2. Configure the version of the tools you want to test.

   ```shell
   # Depending on the versions used, 
   export HELM_VERSION="3.2.2"        # Helm version: for Helm v2.x.x we have to initialise Tiller right after the k8s cluster
   export HELM_CHART_VERSION="7.6.1"  # version of the Elastic's Observability Helm charts
   export KUBERNETES_VERSION="1.15.3" # version of the cluster to be passedd to kind
   ```

3. Install dependencies.

   - Install Go with Gimme: `.ci/scripts/install-go.sh 1.13.4`
   - Configure Go Path:
      - Mac: `export GOROOT=${HOME}/.gimme/versions/go1.13.4.darwin.amd64`
      - Linux: `export GOROOT=${HOME}/.gimme/versions/go1.13.4.linux.amd64`
   - Install godog: `make -C e2e install-godog`

4. Run the tests.

   ```shell
   cd e2e/_suites/helm
   OP_LOG_LEVEL=DEBUG godog
   ```

>    The tests will take a few minutes to run, spinning up the kubernetes cluster, instgalling the helm charts, and performing the test steps outlined earlier.

Reviewer: ```suggestion
   The tests will take a few minutes to run, spinning up the Kubernetes cluster, installing the helm charts, and performing the test steps outlined earlier.
```
